cep-9999.md

```diff
@@ -0,0 +1,171 @@
+<table>
+<tr><td> Title </td><td> Serving run_exports metadata in conda channels </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td>
+<td> 
+    Jaime Rodríguez-Guerra &lt;jrodriguez@quansight.com&gt;
+</td></tr>
+<tr><td> Created </td><td> May 4, 2023</td></tr>
+<tr><td> Updated </td><td> May 4, 2023</td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda-incubator/ceps/pull/51 </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+  "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+  described in [RFC2119][RFC2119] when, and only when, they appear in all capitals, as shown here.
+
+## Abstract
+
+Have conda channels serve standalone `run_exports` metadata, next to `repodata.json`.
+
+## Motivation
+
+Building infrastructure (such as some conda-forge bots) use `run_exports` to calculate which packages need rebuilding as part of an upgrade. Right now, conda-forge needs to maintain their own JSON database, which involves downloading and extracting the new artifacts as they become available: the `run_exports` metadata lives inside the `.tar.bz2` and `.conda` files.
+
+Since `conda-index` already processes the package metadata to generate `repodata.json`, it would be trivial to also generate `run_exports.json` and serve them together.
+
+## Precedence and role in the pinning resolution process
+
+This file is not meant to replace the `run_exports` metadata already present in the package archives. It merely presents that information in a more convenient way. 
+`conda-build`-like clients are free to use either the `run_exports` metadata in the archives or the one in `run_exports.json`, since they MUST be equivalent.
+Special keys like `pin_run_as_build` MUST keep their behavior, since `run_exports.json` does not add a new level of precedence in the pinning resolution process. Again, it's an equivalent source for information already present in the archives.
+
+This means that `run_exports.json` MUST NOT be patched (like it is done with `repodata.json`);
+it MUST always reflect the metadata present in the archives. 
```

Comment:
I wasn't sure about this, because it seemed like recipes could rot and become broken without the repo data patches. I think we are ok, because the repo data patches still apply when the build/host/run environments are created. Is it worthwhile to add a note about this eventual patching here, or is that already clear enough to everyone (except maybe me)?

Reply:
I am not sure I follow. This CEP tries to say that `run_exports.json` won't be patched for the sake of simplicity of scope, but we can always submit a new CEP overriding that decision if we ever arrive to a point where we need run_exports patching.

The previous discussion around it seemed to contain mixed opinions about whether patching run-exports was a good idea or not, so for now we are leaving things as they are.

What kind of wording would help clarify the intent here? I am happy to add it, but changing the intent of the CEP is a different thing given how far into the vote we are already.

Thanks for reviewing!

Reply:
I agree with the decision to not patch it right now, and I think it might be fine to never patch it. The only explanation I would add is that this decision is not ignoring patching, it is deferring it to the environment creation step. If this is shown to create invalid environments, then a follow on CEP would be needed to apply patching here. You could probably prove that one way or the other with a thought experiment now, but I think the incremental approach is a good enough start.

Reply:
Thanks for the explanation! What about adding this sentence:

> Note this only applies to the served `run_exports.json` file. It does not try to regulate what `conda-build`-like tools can do; e.g. they might need to apply modifications to the `run_exports` metadata during the `build` and `host` environments setup.

Reply:
How about:

Note this only applies to the served run_exports.json file. It does not try to regulate what conda-build-like tools can do at environment creation time. They might need to apply modifications analogous to repodata patching to the run_exports metadata during the build, host and test environments setup. If patching run _exports.json is shown to be necessary for correct environment creation, it will be the subject of another CEP.